codecov.yml

```diff
@@ -0,0 +1,109 @@
+# curl --data-binary @codecov.yml https://codecov.io/validate
+codecov:
+  require_ci_to_pass: yes # wait for all other statuses to pass before sending its status
+  branch: main
+  # strict_yaml_branch: main
+
+coverage:
+  precision: 2
+  round: down
+  range: "40...100"         # less than 40% having a red background
+
+comment:
+  layout: "reach,diff,flags"
+  behavior: default         #  update, if exists. Otherwise post new.
+  require_changes: true     # if true: only post the comment if coverage changes
+  require_base: no          # [yes :: must have a base report to post]
+  require_head: yes         # [yes :: must have a head report to post]
+
+flags:
```

Comment:
is there a way to centralise this? I.e. make it part of the Go package definition somehow?
Maybe by introducing a better naming scheme of the coverage file, or by providing this mapping from within leeway so that we can produce it when we run codecov.

Reply:
@csweichel flags is the solution provided by codecov https://docs.codecov.com/docs/flags
Also, I want to switch `strict_yaml_branch: main` to on in a future PR to ensure we don't change the configuration in a PR and only have one valid configuration